src/components/calcite-tile-select/calcite-tile-select.tsx

```diff
@@ -67,6 +67,27 @@ export class CalciteTileSelect {
 
   private input: HTMLCalciteCheckboxElement | HTMLCalciteRadioButtonElement;
 
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private checkFirstTileSelect(): void {
+    const tileSelects = document.querySelectorAll(`calcite-tile-select[name=${this.name}]`);
```

Comment:
Any way to make this more performant so it doesn't have to query the whole document for these elements? Maybe an event listener on each element that listens at document level?

Reply:
There's no need to setup a listener since this function only needs to run on initial mount, and since there's no way to reference sibling elements with JSX, I don't really see an alternative to using `document.querySelectorAll` to find them.

Having said that, since we want to guide users to wrap `tile-select` elements inside `tile-select-group` elements, I can imagine putting this functionality into `tile-select-group`, but the problem with doing that is it removes the ability to use `tile-select` components standalone just like native `input type="radio"` elements, which we need to support since they use them under the hood.

I can imagine a scenario where a developer needs to render `tile-select` elements in a custom layout where the individual `tile-select` elements can't be direct children of `tile-select-group`.  I realized this need early on when I was building `radio-button` to mimic how native radio inputs work, which don't require a native radio group element since it doesn't exist.

Also, I'm sure native `input` elements must have to do something similar to ensure that the first checked radio wins.  Would be cool to know how they actually do that.  Maybe I should peek at the Chromium source code?? 😆

Reply:
yeah makes sense.  👍